### ISSUE TITLE
Inject repository getter on tests

### ISSUE THREAD

**Issue by israelglar** [strongloop/loopback-next#1922]
**TLDR:** How should I retrieve the Getter of a repository to use in testing? 

### Description / Steps to reproduce / Feature proposal
I'm trying to write a Integration tests to my login functionality on my ProfileController. To do that, I'm following what's proposed here: https://loopback.io/doc/en/lb4/Testing-your-application.html#test-controllers-and-repositories-together

My code right now looks like this:
**profile.controller.integration.ts**
```ts
describe('Authentication', async () => {
    let controller: ProfileController;
    let profile: Profile;
    let data: ProfileData;
    before(async () => {
      controller = await profileController();
      ({profile, data} = await givenProfile());
    });
    it('can login', async () => {
      const res = await controller.login({
        username: data.username,
        password: data.password,
      });

      expect(res).to.be.an('object');
      expect(res).to.have.property('id');
    });
    it('can access authenticated route (own profile info)', async () => {
      const res = await controller.findById(profile.id);

      expect(res).to.be.an('object');
      expect(res).to.have.all.keys(profile);
      expect(
        new ObjectId(profile.id).equals(res.id),
        'Received wrong profileId',
      ).to.be.equals(true);
    });
  });
```
**profile.helper.ts**
```ts
import {injectRepoGetter} from './repository.helper';

export async function profileController(): Promise<ProfileController> {
  return new ProfileController(
    new AccessTokenRepository(
      testdb,
      await injectRepoGetter('ProfileRepository'),
    ),
    new ProfileRepository(testdb),
  );
}

export type ProfileData = {
  name: string;
  username: string;
  password: string;
  email: string;
};

export async function givenProfile(data?: Partial<Profile>) {
  const defaultData: ProfileData = {
    name: 'test',
    username: 'test',
    password: 'test',
    email: 'email@test.com',
  };
  const profileData = Object.assign(defaultData, data);
  return {
    data: defaultData,
    profile: await new ProfileRepository(testdb).create(
      new Profile(profileData),
    ),
  };
}
```
**repository.helpers.ts**
```ts
const context_1 = require('@loopback/context');

export function injectRepoGetter(repo: string) {
  return context_1.inject.getter(`repositories.${repo}`);
}
```

The relation between Profile and AccessToken is **AccessToken belongsTo Profile**.
So to instantiate the ProfileController I need to inject the instances of AccessTokenRepository and ProfileRepository.

The constructor of AccessTokenRepository is as follows:
```ts
constructor(
  @inject('datasources.GlarDB') dataSource: GlarDBDataSource,
  @repository.getter('ProfileRepository')
  profileRepositoryGetter: Getter<ProfileRepository>,
) {
  super(AccessToken, dataSource);
  this.user = this._createBelongsToAccessorFor(
    'userId',
    profileRepositoryGetter,
  );
}
```

The problem is where do I retrieve the getter of ProfileRepository in tests? **Getter<ProfileRepository>**

What I did was go trough the code and call the function that injects the Getter directly. But that seems wrong and that it's not supposed to be that way.

Is there a "right" way to retrieve those Getters to use in testing?

### Current Behavior
Directly import the function that retrieves the Getter of a repository to be able to instantiate Controllers and test models that are related to others, 

### Expected Behavior
A "conventional" way to retrieve the Getter of a repository in testing
Not only the repository getters but also the other others "injects" like datasource, repository, AuthenticationBindings.METADATA on authentication providers...

**Comment by raymondfeng** [comment]
The `Getter<T>` is simple function as `() => T`. You can create your owner getter for testing. If you want to use `inject.getter` as a function:
```ts
import {inject} from '@loopback/context';
inject.getter(...)
```

**Comment by israelglar** [comment]
Thanks for the help! I was able do it like this:
**profile.helper.ts**
```
import {Getter} from '@loopback/context';
import {ProfileController} from './../../src/controllers/profile.controller';
import {ProfileRepository, AccessTokenRepository} from '../../src/repositories';
import {testdb} from '../fixtures/datasources/testdb.datasource';

export async function profileController(): Promise<ProfileController> {
  const profileRepository = new ProfileRepository(testdb);
  return new ProfileController(
    new AccessTokenRepository(testdb, Getter.fromValue(profileRepository)),
    profileRepository,
  );
}
```

Indeed it's much simpler. But I think that should be somewhere on the tests documentation.
Thanks!

**Comment by bajtos** [comment]
```ts
Getter.fromValue(profileRepository)
```

☝️ that's the right API to use 👍 

> I think that should be somewhere on the tests documentation.

I agree. Could you please advise us where to put such documentation? What is the place you would be looking (or have been looking) for this information?

Would you mind to contribute this improvement yourself, to help future LoopBack users? See https://github.com/strongloop/loopback-next/blob/master/docs/CONTRIBUTING.md. Unfortunately the section about documentation is outdated, all documentation files are hosted inside loopback-next now (see https://github.com/strongloop/loopback-next/tree/master/docs/site).

**Comment by israelglar** [comment]
I would put somewhere here: https://loopback.io/doc/en/lb4/Testing-your-application.html#integration-testing. Because it's the first time that talks about tests with instantiated Controllers and repositories. 

Also maybe add something on the TodoList example https://github.com/strongloop/loopback-next/tree/master/examples/todo-list

I would love to contribute. I can't today, but tomorrow I can start looking into that :+1:

**Comment by bajtos** [comment]
@israelglar thank you for the suggestions. Have you managed to find time to start working on these improvements? Is there anything we can help you with along the road?

**Comment by edferreira** [comment]
@bajtos in case both both repositories have getter references one to another, like in a hasOne and BelongsTo relation, how should the getter be passed? 
Or maybe shouldn't two repositories have getter referencing each other?

**Comment by bajtos** [comment]
Closing as done by https://github.com/strongloop/loopback-next/pull/2060.
